android/src/main/java/com/marigold/rnsdk/RNEngageBySailthruModule.kt

```diff
@@ -0,0 +1,316 @@
+package com.marigold.rnsdk
+
+import androidx.annotation.VisibleForTesting
+import com.facebook.react.bridge.Promise
+import com.facebook.react.bridge.ReactApplicationContext
+import com.facebook.react.bridge.ReactContextBaseJavaModule
+import com.facebook.react.bridge.ReactMethod
+import com.facebook.react.bridge.ReadableArray
+import com.facebook.react.bridge.ReadableMap
+import com.marigold.sdk.EngageBySailthru
+import com.marigold.sdk.Marigold
+import com.marigold.sdk.model.AttributeMap
+import com.marigold.sdk.model.Purchase
+import org.json.JSONException
+import org.json.JSONObject
+import java.net.URI
+import java.net.URISyntaxException
+import java.util.ArrayList
+import java.util.Date
+
+class RNEngageBySailthruModule (reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext) {
+
+    @VisibleForTesting
+    internal var jsonConverter = JsonConverter()
+
+    @ReactMethod
+    fun logEvent(value: String) {
+        createEngageBySailthru().logEvent(value)
+    }
+
+    @ReactMethod
+    fun logEvent(eventName: String, varsMap: ReadableMap) {
+        var varsJson: JSONObject? = null
+        try {
+            varsJson = jsonConverter.convertMapToJson(varsMap)
+        } catch (e: JSONException) {
+            e.printStackTrace()
+        }
+        createEngageBySailthru().logEvent(eventName, varsJson)
+    }
+
+    @ReactMethod
+    @Deprecated("use setProfileVars instead")
+    fun setAttributes(readableMap: ReadableMap, promise: Promise) {
+        val attributeMap = try {
+            getAttributeMap(readableMap)
+        } catch (e: JSONException) {
+            promise.reject(RNMarigoldModule.ERROR_CODE_DEVICE, e.message)
+            return
+        }
+        createEngageBySailthru().setAttributes(attributeMap, object : EngageBySailthru.AttributesHandler {
+            override fun onSuccess() {
+                promise.resolve(null)
+            }
+
+            override fun onFailure(error: Error) {
+                promise.reject(RNMarigoldModule.ERROR_CODE_DEVICE, error.message)
```

Comment:
Since these error codes are only used in the `RNEngageBySailthruModule` we can probably move them across.